token_dispenser_client/token_dispenser_client.py

```diff
@@ -0,0 +1,216 @@
+_DEFAULT_SSM_PATH_: str = '/service/token-dispenser'  # ssm parameter store default path for TDS ARN
+import boto3
+import json
+import re
+from botocore.exceptions import ClientError
+lambda_client = boto3.client('lambda')
+ssm = boto3.client('ssm')
+
+
+def get_parameter_by_name(name: str, with_decryption:bool = False)->str:
+    try:
+        if not name.endswith('/'):  # Check if it's a specific parameter name
+            response = ssm.get_parameter(
+                Name=name,
+                WithDecryption=with_decryption
+            )
+        return response['Parameter']['Value']
+    except KeyError:
+        print(f"Missing 'Parameter' or 'Value' key with name: {name} response:{response} "
+                       f"with_decryption:{with_decryption}")
+        raise KeyError(f"Missing 'Parameter' or 'Value' key with name: {name} response:{response} "
+                       f"with_decryption:{with_decryption}")
+    except Exception as e:  # Catch any other error.
+        print(f"An unexpected error occurred during get_parameter: name: {name} response:{response} "
+                       f"with_decryption:{with_decryption}")
+        raise e
+
+
+def get_parameters_by_path(path:str, with_decryption=True):
+    """
+    Retrieves parameters from AWS Systems Manager Parameter Store by path.
+
+    Args:
+        path (str): The path to the parameters.
+        with_decryption (bool, optional): Whether to decrypt SecureString parameters. Defaults to True.
+
+    Returns:
+        dict: A dictionary containing the parameter names and values, or None if an error occurs.
+        Returns empty dict if no parameters found on the path.
+    Raises:
+        botocore.exceptions.ClientError: If an error occurs while calling the SSM API.
+    """
+    try:
+        parameters = {}
+        next_token = None
+        while True:
+            if next_token:
+                response = ssm.get_parameters_by_path(
+                    Path=path,
+                    Recursive=True,
+                    WithDecryption=with_decryption,
+                    NextToken=next_token
+                )
+            else:
+                response = ssm.get_parameters_by_path(
+                    Path=path,
+                    Recursive=True,
+                    WithDecryption=with_decryption
+                )
+
+            for param in response.get('Parameters', []):
+                parameters[param['Name']] = param['Value']
+
+            next_token = response.get('NextToken')
+            if not next_token:
+                break
```

Comment:
We don't need to retrieve all parameter values; we just need to retrieve a maximum of 2. If the amount of results is 0 or 2, service discovery failed and the developer has to explicitly define the lambda ARN. If the amount of results is 1, we use that as the lambda ARN